nsiqcppstyle_exe.py

```diff
@@ -371,40 +409,22 @@ def __init__(self, fileFilterPath, extLangMap, varMap, activeFilterName):
         filter = self.GetFilter(self.defaultFilterName)
         self.activeFilterName = self.defaultFilterName
 
+        if filterString:
+            lines = filterString.split(';')
+            for line in lines:
+                filter = self._ProcessFilterLine(filter, line.strip())
```

Comment:
Use `map` instead

Reply:
`map` will not work in this case.  `filter` is a volatile reference.  It begins life as `filter = self.GetFilter('default')`, but can be changed if a filterfile.txt line instructs it to change, e.g.:

```
* FILTER_SCOPE_NAME
```

That's why `self._ProcessFilterLine` in the loop has to be passed the current `filter`, but can be returned a different instance:

```
if filterString:
    lines = filterString.split(';')
    for line in lines:
        filter = self._ProcessFilterLine(filter, line.strip())

f = self.GetFilterFile(fileFilterPath)
if f:
    for line in f.readlines():
        filter = self._ProcessFilterLine(filter, line)
    f.close()
```
Also, `map` does not allow passing a changing argument like `filter`, so we would have to make `filter` a class member variable, complicating the code.  The current code knows that it has a current `filter`, this variable is passed as an argument, and may return the same or a different filter instance.  Once finished we throw the temporary variable away.